dashboard-new/src/app/(authenticated)/wfSpec/[...props]/components/WfRuns.tsx

```diff
@@ -0,0 +1,85 @@
+'use client'
+import { SearchFooter } from '@/app/(authenticated)/components/SearchFooter'
+import { SEARCH_DEFAULT_LIMIT, TIME_RANGES, TimeRange } from '@/app/constants'
+import { useWhoAmI } from '@/contexts/WhoAmIContext'
+import { ArrowPathIcon } from '@heroicons/react/16/solid'
+import { useInfiniteQuery } from '@tanstack/react-query'
+import { LHStatus, lHStatusFromJSON } from 'littlehorse-client/dist/proto/common_enums'
+import { WfRunIdList } from 'littlehorse-client/dist/proto/service'
+import { WfSpec } from 'littlehorse-client/dist/proto/wf_spec'
+import Link from 'next/link'
+import { useSearchParams } from 'next/navigation'
+import { FC, useMemo, useState } from 'react'
+import { searchWfRun } from '../actions/searchWfRun'
+import { WfRunsHeader } from './WfRunsHeader'
+
+type Props = Pick<WfSpec, 'id'>
+export const WfRuns: FC<Props> = ({ id }) => {
+  const searchParams = useSearchParams()
+  const status = getStatus(searchParams.get('status')) || LHStatus.RUNNING
+  const [limit, setLimit] = useState<number>(SEARCH_DEFAULT_LIMIT)
+  const [window, setWindow] = useState<TimeRange>(TIME_RANGES[0])
+  const { tenantId } = useWhoAmI()
+
+  const startTime = useMemo(() => {
+    const now = new Date()
+    const latestStart = now.toISOString()
+    const earliestStart = new Date(now.getTime() - window * 6e4).toISOString()
+
+    return {
+      latestStart,
+      earliestStart,
+    }
+  }, [window])
+
+  const { isPending, data, hasNextPage, fetchNextPage } = useInfiniteQuery({
+    queryKey: ['wfRun', status, tenantId, limit, startTime],
+    initialPageParam: undefined,
+    getNextPageParam: (lastPage: WfRunIdList) => lastPage.bookmark?.toString('base64'),
+    queryFn: async ({ pageParam }) => {
+      console.log({ startTime })
+      return await searchWfRun({
+        wfSpecName: id!.name,
+        wfSpecMajorVersion: id!.majorVersion,
+        wfSpecRevision: id!.revision,
+        variableFilters: [],
```

Comment:
You can use this to search by variables, but it only works for variables in the Entrypoint ThreadRun.

Note that there is also `rpc SearchVariable` which is much more performant, but it requires the variables to be indexed. However it allows variables in any threadRun (so long as the variable has an index)